server/apps/on_post_created.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package apps
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+type PostCreatedNotification struct {
+	SubscriptionID SubscriptionID
+	Subject        SubscriptionSubject
+	ChannelID      string
+	ParentID       string
+	PostID         string
+	RootID         string
+	UserID         string
+	// TeamID         string
+	Expanded *Expanded
+}
+
+// OnUserHasBeenCreated sends a change notification when a new user has
+// joined a team.
+func (p *proxy) OnPostHasBeenCreated(ctx *plugin.Context, post *model.Post) {
```

Comment:
This can be worked into a function shared for all subscriptions, with a handful of parameters.